src/frontend/src/flows/addDevice/welcomeView/registerTentativeDevice.ts

```diff
@@ -1,174 +1,168 @@
-import { html, render } from "lit-html";
+import { html } from "lit-html";
 import { creationOptions, Connection } from "../../../utils/iiConnection";
+import {
+  unreachableLax,
+  unreachable,
+  unknownToString,
+} from "../../../utils/utils";
 import { WebAuthnIdentity } from "@dfinity/identity";
 import { deviceRegistrationDisabledInfo } from "./deviceRegistrationModeDisabled";
-import { DerEncodedPublicKey } from "@dfinity/agent";
 import {
-  KeyType,
-  Purpose,
+  DeviceData,
+  AddTentativeDeviceResponse,
+  CredentialId,
 } from "../../../../generated/internet_identity_types";
 import { showVerificationCode } from "./showVerificationCode";
 import { withLoader } from "../../../components/loader";
-import { toggleErrorMessage } from "../../../utils/errorHelper";
+import { promptDeviceAlias } from "../../../components/alias";
 import { displayError } from "../../../components/displayError";
-import { mainWindow } from "../../../components/mainWindow";
-
-const pageContent = ({ userNumber }: { userNumber: bigint }) => {
-  const pageContentSlot = html` <article>
-    <hgroup>
-      <h1 class="t-title t-title--main">Add a Trusted Device</h1>
-      <p class="t-lead">
-        What device do you want to add to anchor
-        <strong class="t-strong">${userNumber}</strong>?
-      </p>
-      <p id="invalidAliasMessage" class="is-hidden">
-        The device alias must not be empty.
-      </p>
-    </hgroup>
-    <input
-      type="text"
-      id="tentativeDeviceAlias"
-      placeholder="Example: my phone"
-      maxlength="64"
-      class="c-input"
-    />
-    <div class="c-button-group">
-      <button
-        class="c-button c-button--secondary"
-        id="registerTentativeDeviceCancel"
-      >
-        Cancel
-      </button>
-      <button
-        class="c-button"
-        id="registerTentativeDeviceContinue"
-        class="primary"
-      >
-        Continue
-      </button>
-    </div>
-  </article>`;
 
-  return mainWindow({
-    showLogo: false,
-    showFooter: false,
-    slot: pageContentSlot,
-  });
-};
 /**
  * Prompts the user to enter a device alias. When clicking next, the device is added tentatively to the given identity anchor.
  * @param userNumber anchor to add the tentative device to.
  */
-export const registerTentativeDevice = (
+export const registerTentativeDevice = async (
   userNumber: bigint,
   connection: Connection
 ): Promise<void> => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent({ userNumber }), container);
-  return init(userNumber, connection);
-};
+  // First, we need an alias for the device to (tentatively) add
+  const alias = await promptDeviceAlias({
+    title: "Add a Trusted Device",
+    message: html` What device do you want to add to anchor
```

Comment:
> "What device do you want to add to anchor"

"...add to **your** anchor" maybe?

Reply:
There's a `<strong>100512312</strong>` right after that, not evident in the diff though. I tripped on this too 😅